src/authorization/ParentContainerReader.ts

```diff
@@ -0,0 +1,100 @@
+import type { CredentialGroup } from '../authentication/Credentials';
+import type { ResourceIdentifier } from '../http/representation/ResourceIdentifier';
+import { getLoggerFor } from '../logging/LogUtil';
+import type { IdentifierStrategy } from '../util/identifiers/IdentifierStrategy';
+import { IdentifierMap, IdentifierSetMultiMap } from '../util/map/IdentifierMap';
+import { modify } from '../util/map/MapUtil';
+import type { PermissionReaderInput } from './PermissionReader';
+import { PermissionReader } from './PermissionReader';
+import type { PermissionMap, Permission, PermissionSet, AccessMap } from './permissions/Permissions';
+import { AccessMode } from './permissions/Permissions';
+
+/**
+ * Determines `delete` and `create` permissions for those resources that need it
+ * by making sure the parent container has the required permissions.
+ *
+ * Create requires `append` permissions on the parent container.
+ * Delete requires `write` permissions on both the parent container and the resource itself.
+ */
+export class ParentContainerReader extends PermissionReader {
+  protected readonly logger = getLoggerFor(this);
+
+  private readonly reader: PermissionReader;
+  private readonly identifierStrategy: IdentifierStrategy;
+
+  public constructor(reader: PermissionReader, identifierStrategy: IdentifierStrategy) {
+    super();
+    this.reader = reader;
+    this.identifierStrategy = identifierStrategy;
+  }
+
+  public async handle({ requestedModes, credentials }: PermissionReaderInput): Promise<PermissionMap> {
+    const containerMap = this.findParents(requestedModes);
+
+    const updatedMap = modify(new IdentifierSetMultiMap(requestedModes), { add: containerMap.values() });
```

Comment:
```suggestion
    const containerModes = this.findParents(requestedModes);
    const allModes = modify(new IdentifierSetMultiMap(requestedModes), { add: containerMap.values() });
```
(or `combinedModes` or so)